tests/integration/test_charm.py

```diff
@@ -68,3 +75,85 @@ async def test_owasp_modsecurity_custom_rules(app_url_modsec_ignore: str):
     """
     response = requests.get(app_url_modsec_ignore)
     assert response.status_code == 200
+
+
+async def test_owasp_modsecurity_crs_relation(ops_test: OpsTest, app_name: str, tmp_path: Path):
+    """
+    arrange: given charm has been built and deployed
+    act: relate the ingress integrator with any-charm and use any-charm to toggle modsecurity
+        option through relation via ingress lib.
+    assert: modsecurity annotations should be attached and detached from the kubernetes ingress
+        resource according to the modsecurity option.
+    """
+    await ops_test.model.applications[app_name].set_config(
+        {"owasp-modsecurity-crs": "false", "owasp-modsecurity-custom-rules": ""}
+    )
+    any_charm_tmp_path = tmp_path / "any-charm"
+    if not any_charm_tmp_path.exists():
+        subprocess.run(  # nosec
+            ["git", "clone", "https://github.com/canonical/any-charm.git", any_charm_tmp_path]
+        )
+
+    any_charm = await ops_test.build_charm(any_charm_tmp_path)
+    ingress_lib = Path("lib/charms/nginx_ingress_integrator/v0/ingress.py").read_text()
+    any_charm_src_overwrite = {
+        "ingress.py": ingress_lib,
+        "any_charm.py": dedent(
```

Comment:
Another option here is to create a fixture file (like `any_charm.py`) and just read it from disk, it would mean that we get nice Python formatting when looking at the `any_charm.py` file and don't need to `dedent`. What do you think?

Reply:
Yes, I will put it in a separated file.